### ISSUE TITLE
Modified it a bit

### ISSUE THREAD

**Issue by Brutus11111** [amrut-asm/pselc#1]
I'm supposed to do a pull request but I got no idea how that's supposed to work. I forked it and added zathura and an option to quit it without using ctrl+c

Good luck 

https://github.com/Brutus11111/pselc/blob/master/pselc

**Comment by amrut-asm** [comment]
> I'm supposed to do a pull request but I got no idea how that's supposed to work. I forked it and added zathura and an option to quit it without using ctrl+c
> 
> Good luck
> 
> https://github.com/Brutus11111/pselc/blob/master/pselc
Thanks. I added zathura support.

However, I'm not sure about the quit option. Because, let's say an user enters an invalid number. Then, he should get a second chance in case the input error was unintentional.

**Comment by Brutus11111** [comment]
q for quit is indeed what I wanted. Thanks for adding it. 

I was also thinking, why not specify the pdf reader in .config/pselc/config, so you can manually set it. Right now, it just grabs the last one it finds. If I put Zathura first, it will overwrite it with another PDF reader if I have one installed.

**Comment by scoronado12** [comment]
So here's my thought, because there are a ridiculous amount of PDF viewers out in the wild.

You could theoretically make use of `xdg-open` which will read from the system what is your default pdf viewer

Usage would be such as
`$ xdg-open resume.odt`

Which would open up LibreOffice Writer or if I did:
`$xdg-open resume.pdf`

It would open up Okular by default.

**Comment by amrut-asm** [comment]
> q for quit is indeed what I wanted. Thanks for adding it.
> 
> I was also thinking, why not specify the pdf reader in .config/pselc/config, so you can manually set it. Right now, it just grabs the last one it finds. If I put Zathura first, it will overwrite it with another PDF reader if I have one installed.

The script has a variable named "READER" at line 4. You can set it to your preferred PDF reader and it'll work with that.

I don't know about the .config/pselc part because I'm quite new to BASH.

**Comment by amrut-asm** [comment]
> So here's my thought, because there are a ridiculous amount of PDF viewers out in the wild.
> 
> You could theoretically make use of `xdg-open` which will read from the system what is your default pdf viewer
> 
> Usage would be such as
> `$ xdg-open resume.odt`
> 
> Which would open up LibreOffice Writer or if I did:
> `$xdg-open resume.pdf`
> 
> It would open up Okular by default.

Yes, that's what I did earlier but on some distros (like mine, Gentoo) you would have to manually set the PDF reader (with "xdg-mime default") then call xdg-open. This is not usually the case on most distros but still, the script might not work for some people in these type of weird cases.

If the user has more than one PDF readers on their system, he/she can simply replace the "READER" variable (line 4 in the script) to point to their preferred PDF reader.

**Comment by scoronado12** [comment]
> Yes, that's what I did earlier but on some distros (like mine, Gentoo) you would have to manually set the PDF reader (with "xdg-mime default") then call xdg-open. This is not usually the case on most distros but still, the script might not work for some people in these type of weird cases.

That looks goofy (to me - who has never used Gentoo) but we could also theoretically parse `/etc/os-release` or part of the line of the running kernel (similar to how `neofetch` does it) and detect whether or not we need to do that.

Again, it is just a thought... the script looks like it could be really helpful on a keyboard driven experience.

**Comment by amrut-asm** [comment]
> So here's my thought, because there are a ridiculous amount of PDF viewers out in the wild.
> 
> You could theoretically make use of `xdg-open` which will read from the system what is your default pdf viewer
> 
> Usage would be such as
> `$ xdg-open resume.odt`
> 
> Which would open up LibreOffice Writer or if I did:
> `$xdg-open resume.pdf`
> 
> It would open up Okular by default.

Sorry for the late commit. xdg-open is now used by default. This method is definitely superior. I didn't know what I was thinking when I decided to use those nasty if-else statements for finding PDF readers installed.
